### ISSUE TITLE
Add download for "slice"

### ISSUE THREAD

**Issue by cbmdfc** [alexellis/arkade#884]
Hello all,

i'm looking to see if arkade can support installing `kubectl-slice` from @patrickdappollonio: https://github.com/patrickdappollonio/kubectl-slice

i could add this myself but dont know how to proceed,, if you have an example i can use i can try

spasiba!

**Comment by alexellis** [comment]
Hi @cbmdfc do you mind filling out the issue template (I guess you may have deleted it and thought it wasn't something we used) - we do really want that info.

Thanks

Alex

**Comment by alexellis** [comment]
Thanks for doing that. We would welcome a contribution for kubectl-slice 

I'd suggest looking for an old PR and copying the approach - from the PR template, to the testing, to the way the template is formed.

Alex

**Comment by alexellis** [comment]
https://github.com/alexellis/arkade/pull/842

https://github.com/alexellis/arkade/pull/877

If you'd like to sponsor me via GitHub, I can add this for you.

**Comment by patrickdappollonio** [comment]
Hey @cbmdfc @alexellis, cool to see this here and didn't think someone would've been interested in allowing for more download sources!

@cbmdfc I'm happy to help out here as well! Let me know if you want to do it (since it seems you wanted to from the get-go) or I can send a PR probably EOD Eastern Time today.

@alexellis if you would prefer the sponsor, I'm also happy to be a fly on the wall in this PR (and a point of contact if needed moving forward).

Since there's no "namespacing", what would the name of the download be? `kubectl-slice` or `slice`? We had the same issue with `krew`, and the maintainers were kind enough to allow us to own just `slice`.

**Comment by alexellis** [comment]
I think slice makes sense unless there's a common Linux binary with the same name?

Happy to take a PR - just check out the examples I shared. 

And I always welcome more GitHub sponsors to support the time on arkade 👍

**Comment by patrickdappollonio** [comment]
I don't think there's a Linux command with the same name but I could be wrong.

In terms of behaviour, the thing is, to be a `kubectl` plugin, the downloaded binary must be named `kubectl-${PLUGIN_NAME}`. Storing `slice` as `kubectl-slice` allows you to call it both via:

```bash
kubectl slice -f ... 
kubectl-slice -f ... # note the dash here
```

I think it's possible for it to be a symlink too (so have `slice` be the binary name and `kubectl-slice` be a symlink to the binary).

Could there be a mismatch between the downloaded binary name and the command in arkade? Is there support for symlinks (so people can use both `slice` as a standalone binary but if they were to have `kubectl`, they can use it as a plugin as well)?

**Comment by patrickdappollonio** [comment]
I have added a PR for `slice`. Let me know if there's anything else I can help with.

Thanks a lot for the request @cbmdfc!

**Comment by alexellis** [comment]
Yes @patrickdappollonio on the PR immediately after you raised it, our bot commented back to you with instructions.

Can you review please?

**Comment by cbmdfc** [comment]
appreciate the diligence but @alexellis it seems to me if the owner of the application couldnt have his pr merged how can another contributor request permission to @patrickdappollonio to add this tool if the owners own pr was rejected on an email basis

without @patrickdappollonio's approval i wouldn't try myself to send a pr and i'm not sure the "usual contributors" might have his authorization

**Comment by patrickdappollonio** [comment]
If it isn't me the one submitting it, I don't really mind who puts it here. The reason I was doing it was because you asked but also because then I can track this as another dependency of work when pushing updates.

If someone else were to add it, their maintenance now depends on them (which I would tremendously appreciate it 😁!). This shouldn't be a problem since the release is quite standard but I've taken the liberty to make changes that then need to be propagated. The last one was renaming the release names which would clearly break `arkade`'s registry but I would've been happy to maintain the same way as I maintain the record in `krew`.

If you want to add it @cbmdfc feel free to go right ahead, the same goes for @alexellis or any other maintainer.
